drivers/staging/vc04_services/bcm2835-isp/bcm2835-v4l2-isp.c

```diff
@@ -722,10 +724,38 @@ static int bcm2835_isp_s_ctrl(struct v4l2_ctrl *ctrl)
 				    sizeof(struct bcm2835_isp_custom_ccm));
 		break;
 	case V4L2_CID_USER_BCM2835_ISP_LENS_SHADING:
-		ret = set_isp_param(node, MMAL_PARAMETER_LENS_SHADING_OVERRIDE,
-				    ctrl->p_new.p_u8,
-				    sizeof(struct bcm2835_isp_lens_shading));
+	{
+		struct bcm2835_isp_lens_shading ls;
+		struct dma_buf *dmabuf;
+		void *vcsm_handle;
+
+		memcpy(&ls, ctrl->p_new.p_u8,
+		       sizeof(struct bcm2835_isp_lens_shading));
+
+		dmabuf = dma_buf_get(ls.dmabuf);
+		if (!dmabuf)
+			return -EINVAL;
+
```

Comment:
Should this error return be ENOMEM ? No idea what the correct thingy would be.

Reply:
Or is it a allocated handle already? Hmm, think it probably is.

Reply:
This is taking a dmabuf allocated by something else (most likely a dmaheap), passing it in to the ISP driver, and then the ISP driver doing the vcsm calls to import it into the VPU heap.
As such the handle is a parameter into this control, so EINVAL is a sensible return value if it isn't a suitable dmabuf.